editor.planx.uk/src/@planx/components/NumberInput/model.ts

```diff
@@ -52,6 +54,19 @@ export const numberInputValidationSchema = (input: NumberInput) =>
         }
         return value === "0" ? true : Boolean(parseNumber(value));
       },
+    })
+    .test({
+      name: "check for a whole number",
+      message: "Enter a whole number",
+      test: (value: string | undefined) => {
+        if (!value) {
+          return false;
+        }
+        if (input.onlyWholeNumbers && !Number.isInteger(Number(value))) {
```

Comment:
I tried a few different ways of validating the whole number. Using the `Number.isInteger()` seemed the most relevant to what I wanted to check, rather than a string check for a decimal point or anything else.

Reply:
`Number.isInteger()` makes sense to me !